docs: update tutorials and how-tos to use pytest-jubilant (#2406)
canonical/operator: docs/tutorial/write-your-first-machine-charm.md

Integration tests are an important way to check that your charm works correctly when deployed. In contrast to unit tests, integration tests require Juju to be available, and events aren't simulated.

> When you created the initial version of your charm, Charmcraft included integration tests. The tests use the [`pytest-jubilant`](https://github.com/canonical/pytest-jubilant) plugin, which provides a `juju` fixture backed by {external+jubilant:doc}`Jubilant <index>`. We'll expand the tests to cover more of your charm's functionality.

Author: ```suggestion
When you created the initial version of your charm, Charmcraft included integration tests. The tests use {external+jubilant:doc}`Jubilant <index>` to interact with Juju. The [`pytest-jubilant`](https://github.com/canonical/pytest-jubilant) plugin provides the `juju` fixture used in the tests. We'll expand the tests to cover more of your charm's functionality.
```
